core/src/main/java/org/apache/brooklyn/core/location/access/PublicNetworkFaceEnricher.java

```diff
@@ -0,0 +1,373 @@
+                return Maybe.absent();
+            }
+            URI result;
+            try {
+                result = new URI(uri.getScheme(), uri.getUserInfo(), publicTarget.getHostText(), publicTarget.getPort(), uri.getPath(), uri.getQuery(), uri.getFragment());
+            } catch (URISyntaxException e) {
+                LOG.debug("Error transforming URI "+uri+", using target "+publicTarget+"; rethrowing");
+                throw Exceptions.propagateAnnotated("Error transforming URI "+uri+", using target "+publicTarget, e);
+            }
+            return Maybe.of(result.toString());
+        } else {
+            LOG.debug("sensor mapper not transforming URI "+uri+" because no port defined");
+            return Maybe.absent();
+        }
+    }
+
+    protected Maybe<String> transformHostAndPort(Entity source, MachineLocation machine, String sensorVal) {
+        HostAndPort hostAndPort = HostAndPort.fromString(sensorVal);
+        if (hostAndPort.hasPort()) {
+            int port = hostAndPort.getPort();
+            HostAndPort publicTarget = getPortForwardManager().lookup(machine, port);
+            if (publicTarget == null) {
+                LOG.debug("network-facing enricher not transforming {} host-and-port {}, because no port-mapping for {}", new Object[] {source, sensorVal, machine});
+                return Maybe.absent();
+            }
+            return Maybe.of(publicTarget.toString());
+        } else {
+            LOG.debug("network-facing enricher not transforming {} host-and-port {} because defines no port", source, hostAndPort);
+            return Maybe.absent();
+        }
+    }
+
+    protected Maybe<String> transformPort(Entity source, MachineLocation machine, int sensorVal) {
+        if (Networking.isPortValid(sensorVal)) {
+            HostAndPort publicTarget = getPortForwardManager().lookup(machine, sensorVal);
+            if (publicTarget == null) {
+                LOG.debug("network-facing enricher not transforming {} host-and-port {}, because no port-mapping for {}", new Object[] {source, sensorVal, machine});
+                return Maybe.absent();
+            }
+            return Maybe.of(publicTarget.toString());
+        } else {
+            LOG.debug("network-facing enricher not transforming {} port {} because not a valid port", source, sensorVal);
+            return Maybe.absent();
+        }
+    }
+
+    protected Maybe<MachineLocation> getMachine() {
+        return Machines.findUniqueMachineLocation(entity.getLocations());
+    }
+    
+    protected PortForwardManager getPortForwardManager() {
+        PortForwardManager portForwardManager = config().get(PORT_FORWARD_MANAGER);
+        if (portForwardManager == null) {
+            portForwardManager = (PortForwardManager) getManagementContext().getLocationRegistry().getLocationManaged("portForwardManager(scope=global)");
```

Comment:
`portForwardManager(scope=global)` is used at so many places now, that it deserves to be promoted to a constant.